### PR TITLE
fix validate-chart scripts

### PR DESCRIPTION
**Pull request by thomasferrandiz**
- clean-up temp yaml file
- fix k8s version check
- fix rke2-coredns templating

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: scripts/validate-charts [hunk 1]
```diff
@@ -51,7 +51,7 @@ check_system_registry() {
     chart_tmp=$3
 
     yaml_tmp=$(mktemp --suffix .yaml)
-    values="global.systemDefaultRegistry=my-registry,global.cattle.systemDefaultRegistry=my-registry,vCenter.clusterId=test-id"
+    values="global.systemDefaultRegistry=my-registry,global.cattle.systemDefaultRegistry=my-registry,vCenter.clusterId=test-id,global.clusterDNS=10.43.0.10"
```
Comment (brandond) [inline scripts/validate-charts:54]: Can we validate this with dual-stack? Ref: https://github.com/rancher/rke2/issues/5156
```suggestion
    values="global.systemDefaultRegistry=my-registry,global.cattle.systemDefaultRegistry=my-registry,vCenter.clusterId=test-id,global.clusterDNS=10.43.0.10,2001:cafe:43::a"
```

File: scripts/validate-charts [hunk 2]
```diff
@@ -110,7 +113,7 @@ check_airgap() {
     chart_tmp=$3
 
     yaml_tmp=$(mktemp --suffix .yaml)
-    values="vCenter.clusterId=test-id"
+    values="vCenter.clusterId=test-id,global.clusterDNS=10.43.0.10"
```
Comment (brandond) [inline scripts/validate-charts:116]: same as above

**Review by brandond — CHANGES_REQUESTED** [review]

File: scripts/validate-charts (hunk 1 above)
Comment (thomasferrandiz) [reply scripts/validate-charts]: done

File: scripts/validate-charts (hunk 2 above)
Comment (thomasferrandiz) [reply scripts/validate-charts]: done

**Review by brandond — APPROVED** [review]

**Review by rbrtbnfgl — APPROVED** [review]

**Pull request merged**